pkg/kubectl/cmd/util/factory.go

```diff
@@ -106,8 +108,23 @@ func NewFactory(optionalClientConfig clientcmd.ClientConfig) *Factory {
 		clientConfig = DefaultClientConfig(flags)
 	}
 
-	clients := NewClientCache(clientConfig)
+	clients := newClientCache(clientConfig)
 
+	var experimentalClient *client.ExperimentalClient
+	cfg, experimentalClientErr := clientConfig.ClientConfig()
```

Comment:
Ok, sorry.  I still don't like init'ing it up front but I can live with
that particular part for now.

On Mon, Aug 10, 2015 at 2:56 PM, Muhammed Uluyol <notifications@github.com>
wrote:

> In pkg/kubectl/cmd/util/factory.go
> <https://github.com/GoogleCloudPlatform/kubernetes/pull/12405#discussion_r36670130>
> :
>
> >
> > +	var experimentalClient *client.ExperimentalClient
> > +	cfg, experimentalClientErr := clientConfig.ClientConfig()
>
> Maybe I'm not seeing it, but how does Factory creation fail? There is no
> if err != nil { return err } statement here. We are keeping the error and
> returning it when an experimental client is requested, not when the factory
> is created.
>
> If client config panic's then that changes things, but it really should
> not panic.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/GoogleCloudPlatform/kubernetes/pull/12405/files#r36670130>
> .
>



-- 
Clayton Coleman | Lead Engineer, OpenShift